programming/java/challenges/run_test.sh

```
@@ -0,0 +1,31 @@
+echo "Compiling Java files..."
+javac -d "$BIN_DIR" -cp "$LIB_DIR/*" "$SRC_DIR"/HelloWorld/HelloWorld.java "$SRC_DIR"/HelloWorld/HelloWorldTest.java
+if [ $? -ne 0 ]; then
+  echo "Compilation failed!"
+  exit 1
+fi
+echo "Compilation successful!"
+
+# Step 2: Run JUnit tests
+echo "Running tests..."
+java -cp "$BIN_DIR:$LIB_DIR/*" org.junit.platform.console.ConsoleLauncher --select-class HelloWorld.HelloWorldTest
+if [ $? -ne 0 ]; then
```

Shellcheck (0.10.0)

[style] Check exit code directly with e.g. 'if ! mycmd;', not indirectly with $?.

(SC2181)